RealmWeaver/ModuleWeaver.cs

```diff
@@ -230,8 +230,18 @@ private void WeaveType(TypeDefinition type, Dictionary<string, Tuple<MethodRefer
             }
         }
 
+        var objectConstructor = type.GetConstructors()
+            .SingleOrDefault(c => c.Parameters.Count == 0 && c.IsPublic && !c.IsStatic);
+        if (objectConstructor == null)
+        {
+            var nonDefaultConstructor = type.GetConstructors().First();
```

Comment:
This will never fail unless someone royally messes up the IL so I wouldn't worry.